From 1ce0129f22b1d062b93fd303259882910b781326 Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Mon, 29 Sep 2014 00:59:14 -0400
Subject: [virt] hyperv: Fix error return code in netvsc_init_buf()

Message-id: <1411952358-34104-31-git-send-email-jasowang@redhat.com>
Patchwork-id: 96779
O-Subject: [RHEL7.1 PATCH 30/34] hyperv: Fix error return code in netvsc_init_buf()
Bugzilla: 1146357
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>

Bugzilla: 1146357

Fix to return -ENOMEM from the kalloc error handling
case instead of 0.

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Reviewed-by: Haiyang Zhang <haiyangz@microsoft.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit dd1d3f8f9920926aa426589e542eed6bf58b7354)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/hyperv/netvsc.c b/drivers/net/hyperv/netvsc.c
index b96e7f6..8ffc41f 100644
--- a/drivers/net/hyperv/netvsc.c
+++ b/drivers/net/hyperv/netvsc.c
@@ -379,8 +379,10 @@ static int netvsc_init_buf(struct hv_device *device)
 
  net_device->send_section_map =
   kzalloc(net_device->map_words * sizeof(ulong), GFP_KERNEL);
- if (net_device->send_section_map == NULL)
+ if (net_device->send_section_map == NULL) {
+  ret = -ENOMEM;
   goto cleanup;
+ }
 
  goto exit;
 
-- 
1.7.1